### DISCUSSION TITLE
Problem with private registry

### DISCUSSION THREAD

**Discussion by idleberg** [denoland/deno#26384] [Q&A]
Using Deno 2.0, I'm trying to install packages from a private NPM registry (AWS CodeArtifact). I don't think I missed anything from the documentation, but somehow, Deno is unable to find the packages.

Here is my `.npmrc` file:

```shell
# third-party packages will be consumed from npmjs.org
registry=https://registry.npmjs.org/

# private packages will be consumed from AWS CodeArtifact
@my-org:registry=https://my-registry.d.codeartifact.eu-central-1.amazonaws.com/npm/npm/
//my-registry.d.codeartifact.eu-central-1.amazonaws.com/npm/npm/:always-auth=true
//my-registry.d.codeartifact.eu-central-1.amazonaws.com/npm/npm/:_authToken=${CODEARTIFACT_AUTH_TOKEN}
```

And here the result when trying to run a script that has a dependency from that registry:

```sh
$ deno run -A build

error: Error getting response at https://my-registry.d.codeartifact.eu-central-1.amazonaws.com/npm/npm/@my-org/commitlint-config for package "@my-org/commitlint-config": Bad response: 400

{"success":false,"error":"Bad request. The package name '@my-org' is invalid."}
```

Additionally, I have tried several other things:

- use `DENO_NPM_REGISTRY`, which I came across in [this issue](https://github.com/denoland/deno/issues/16105#issuecomment-1335948130)
- create an import map that explicitly maps `@my-org/commitlint-config` to `npm:@my-org/commitlint-org`
- the output of running the same command with `-L debug` doesn't seem to reveal anything useful, but I can post it if it helps

Since I did not find helpful answers on Google or in this repo, I'm reaching out to the community. Hopefully, someone can give me further clues about what I'm missing.

**Comment by marvinhagemeister** [comment]
That's a bug in Deno. Opened an issue for that here: https://github.com/denoland/deno/issues/26385

**Comment by idleberg** [comment]
Fixed & merged in https://github.com/denoland/deno/pull/26390. Thanks @marvinhagemeister for looking into it so quickly!
